### ISSUE TITLE
Null byte from binary string sent in JSON response

### ISSUE THREAD

**Issue by muirdm** [microsoft/pyright#2051]
**Describe the bug**
Pyright can send null byte in JSON response.

**To Reproduce**
```python
a: int = b"\0"
```

**Expected behavior**
Diagnostic generated with "valid" JSON. Null bytes may be technically allowed (not sure), but it doesn't seem like a good idea to send them.

**Additional context**
Error I see in Emacs:
```
Warning (lsp-mode): Failed to parse the following chunk:
’’’
Content-Length: 526

{"jsonrpc":"2.0","method":"textDocument/publishDiagnostics","params":{"uri":"file:///Users/muirdm/scratch/foo.py","diagnostics":[{"range":{"start":{"line":3,"character":9},"end":{"line":3,"character":14}},"message":"Expression of type \"Literal[b'\u0000']\" cannot be assigned to declared type \"int\"\n����\"Literal[b'\u0000']\" is incompatible with \"int\"","severity":1,"code":"reportGeneralTypeIssues","source":"Pyright","codeDescription":{"href":"https://github.com/microsoft/pyright/blob/main/docs/configuration.md"}}]}}
’’’
with message (json-parse-error \u0000 is not allowed without JSON_ALLOW_NUL <callback> 1 357 359) Disable showing Disable logging
```

/cc @yyoncho

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.155, which I just published.
